esti/multipart_test.go

```diff
@@ -56,6 +56,77 @@ func TestMultipartUpload(t *testing.T) {
 	}
 }
 
+func TestMultipartUploadAbort(t *testing.T) {
+	ctx, _, repo := setupTest(t)
+	defer tearDownTest(repo)
+
+	t.Run("exists", func(t *testing.T) {
+		const objPath = mainBranch + "/multipart_file1"
+		createInput := &s3.CreateMultipartUploadInput{
+			Bucket: aws.String(repo),
+			Key:    aws.String(objPath),
+		}
+		createResp, err := svc.CreateMultipartUploadWithContext(ctx, createInput)
+		require.NoError(t, err, "CreateMultipartUpload")
+
+		abortInput := &s3.AbortMultipartUploadInput{
+			Bucket:   aws.String(repo),
+			Key:      aws.String(objPath),
+			UploadId: createResp.UploadId,
+		}
+		_, err = svc.AbortMultipartUploadWithContext(ctx, abortInput)
+		require.NoError(t, err, "AbortMultipartUploadWithContext")
+	})
+
+	t.Run("unknown_upload_id", func(t *testing.T) {
+		const objPath = mainBranch + "/multipart_file2"
+		createInput := &s3.CreateMultipartUploadInput{
+			Bucket: aws.String(repo),
+			Key:    aws.String(objPath),
+		}
+		createResp, err := svc.CreateMultipartUploadWithContext(ctx, createInput)
+		require.NoError(t, err, "CreateMultipartUpload")
+
+		uploadID := aws.StringValue(createResp.UploadId)
+		// reverse the upload id to get valid unknown upload id
+		unknownUploadID := reverse(uploadID)
```

Comment:
🤯

Reply:
wanted valid length and valid characters - but invalid upload id :)